src/celestia/view.cpp

```diff
@@ -261,16 +260,15 @@ View::drawBorder(Overlay* overlay, int gWidth, int gHeight, const Color &color,
 
 
 void
-View::updateFBO(int gWidth, int gHeight)
+View::updateFBO(int gWidth, int gHeight, FramebufferObject::AttachmentType colorAttachment, FramebufferObject::AttachmentType depthAttachment)
 {
     auto newWidth = static_cast<GLuint>(width * gWidth);
     auto newHeight = static_cast<GLuint>(height * gHeight);
-    if (fbo && fbo.get()->width() == newWidth && fbo.get()->height() == newHeight)
+    if (fbo && fbo->width() == newWidth && fbo->height() == newHeight && fbo->colorAttachmentType() == colorAttachment && fbo->depthAttachmentType() == depthAttachment)
```

Comment:
no need for additional checks. we can use either textures or renderbuffers but not switch them.